src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java

```diff
@@ -35,51 +48,140 @@
 import org.springframework.cloud.deployer.spi.core.AppDefinition;
 import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
 import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
 
 /**
+ * Unit tests for the {@link CloudFoundryAppDeployer}.
+ *
  * @author Greg Turnquist
  */
 public class CloudFoundryAppDeployerTests {
 
-    CloudFoundryOperations operations;
+	CloudFoundryOperations operations;
+
+	CloudFoundryClient client;
+
+	Applications applications;
+
+	ApplicationsV2 applicationsV2;
+
+	CloudFoundryAppDeployer deployer;
+
+	@Before
+	public void setUp() {
+
+		operations = mock(CloudFoundryOperations.class);
+		client = mock(CloudFoundryClient.class);
+		applications = mock(Applications.class);
+		applicationsV2 = mock(ApplicationsV2.class);
+		deployer = new CloudFoundryAppDeployer(new CloudFoundryDeployerProperties(), operations, client);
+	}
+
+	@Test
+	public void shouldSwitchToSimpleDeploymentIdWhenGroupIsLeftOut() {
+
+		// given
+		given(operations.applications()).willReturn(applications);
+		given(applications.get(any())).willReturn(Mono.just(ApplicationDetail.builder().build()));
+
+		// when
+		String deploymentId = deployer.deploy(new AppDeploymentRequest(
+				new AppDefinition("test", Collections.emptyMap()),
+				new FileSystemResource("")));
+
+		// then
+		assertThat(deploymentId, equalTo("test"));
+	}
+
+	@Test
+	public void shouldNamespaceTheDeploymentIdWhenAGroupIsUsed() {
+
+		// given
+		given(operations.applications()).willReturn(applications);
+		given(applications.get(any())).willReturn(Mono.just(ApplicationDetail.builder().build()));
+
+		// when
+		String deploymentId = deployer.deploy(new AppDeploymentRequest(
+				new AppDefinition("test", Collections.emptyMap()),
+				new FileSystemResource(""),
+				Collections.singletonMap(AppDeployer.GROUP_PROPERTY_KEY, "prefix")));
+
+		// then
+		assertThat(deploymentId, equalTo("prefix-test"));
+	}
+
+	@Test
+	public void shouldMoveSomeIntoSpringAppJson() throws InterruptedException, JsonProcessingException {
+
+		// given
+		CloudFoundryDeployerProperties properties = new CloudFoundryDeployerProperties();
+
+		// Define the env variables for the app
+		Map<String, String> appDefinitionProperties = new HashMap<>();
+
+		final String fooKey = "spring.cloud.foo";
+		final String fooVal = "this should end up in SPRING_APPLICATION_JSON";
+
+		final String barKey = "spring.cloud.bar";
```

Comment:
Maybe this can not have the 'spring' prefix to emphasize the point that all app defintion props go into SAJ.